### DISCUSSION TITLE
Get Balance Chart Data from Rest API

### DISCUSSION THREAD

**Discussion by victorbalssa** [firefly-iii/firefly-iii#6459] [Firefly III] [answered]
Hello @JC5,

I would like to re-create a specific chart in my app like this: 

<img width="336" alt="image" src="https://user-images.githubusercontent.com/12813321/192004477-648dca3e-64ff-4726-807c-93575db6f11c.png">

It shows the balance for each month for a given currency.

I can see that in the BackOffice the data comes from the request: 
* `GET /chart/report/operations/:accountsList/:start/:end` (`/chart/report/operations/1,2,5,3,4/20220101/20220331`)

Are there API endpoints to fetch the same response e.g. `Earned` & `Spend` by months?

Do you know if I can fetch this endpoint `GET /chart/report/operations/:accountsList/:start/:end` only with an Oauth access token?

Thank you for this amazing application, It is always a pleasure to navigate through the ocean of my financial data from the last 10 years 🌊

**Comment by JC5** [comment]
Not yet, but these are on my list!

**Comment by victorbalssa** [comment]
Hey @JC5,

I wanted to know if I could create a PR to support this endpoint in a new API endpoint (v1) and the requirements behind this PR (naming conventions, etc) ?

I have the frontend ready on https://github.com/victorbalssa/abacus.

(https://github.com/firefly-iii/firefly-iii/pull/7161)

Thanks a lot!
Victor

**Reply by JC5** [reply, marked as answer]
Sorry, other efforts are taking up more time than I thought. I know this is an important requirement for your app. Is it OK if I spend some time on this over the next few days?

**Reply by victorbalssa** [reply]
More than OK,
I have a global filter on currency, if the currency `code` could be part of this endpoint query params, it would be helpful.
However, as this can be time-consuming, converting the existing access point into an API access point is more than sufficient.

Thank you for considering this option!
